js/src/tags.js

```diff
@@ -0,0 +1,467 @@
+        delete this.modified_items[id];
+        delete this.item_errors[id];
+
+        if (this.deleted_items_ids.includes(id)) {
+            this.deleted_items_ids.splice(this.deleted_items_ids.indexOf(id));
+        }
+        if (this.new_item_ids.includes(id)) {
+            this.getModifiedItem(id);
+        }
+    };
+
+    ListableItemsAdmin.prototype.save = function () {
+        var self = this;
+
+        self.item_errors = {};
+
+        var new_item_ids = self.new_item_ids;
+        var modified_items = self.modified_items;
+        var url = self.saveUrl;
+
+        var body = {
+            created: {},
+            modified: {},
+            deleted: self.deleted_items_ids
+        };
+
+        for (var id in modified_items) {
+            if (new_item_ids.includes(id)) {
+                body.created[id] = modified_items[id];
+            } else {
+                body.modified[id] = modified_items[id];
+            }
+        }
+
+        $.ajax({
+            url: url,
+            type: 'POST',
+            dataType: 'json',
+            contentType: 'application/json',
+            data: JSON.stringify(body)
+        }).done(function (response) {
+            if ('errors' in response) {
+                self.item_errors = response.errors;
+            }
+
+            for (var id in self.loaded_items) {
+                if (!(id in self.item_errors)) {
+                    self.forgetItem(id);
+                }
+            }
+
+            self.changePage(1);
+            self.loadPageCount();
+        });
+    };
+
+    ListableItemsAdmin.prototype.refreshDisplay = function () {
+        this.items_list.empty();
+
+        for (var offset in this.column_headings) {
+            var heading = $($.parseHTML(_.template(this.template_heading.html())({})));
+            heading.append(this.column_headings[offset]);
+            this.items_list.append(heading);
+        }
+
+        if (Object.keys(this.item_errors).length > 0) {
+            this.errors_banner.show();
+        } else {
+            this.errors_banner.hide();
+        }
+
+        var displayed_ids = this.displayed_ids;
+
+        if (displayed_ids.length === 0) {
+            this.no_results_banner.show();
+        } else {
+            this.no_results_banner.hide();
+            for (var offset in displayed_ids) {
+                var item = this.getItem(displayed_ids[offset]);
+                this.items_list.append(this.displayRow(displayed_ids[offset], item));
+            }
+        }
+
+        this.checkModified();
+    };
+
+    ListableItemsAdmin.prototype.displayRow = function (id, item) {
+        var new_item = this.new_item_ids.includes(id);
+        var edit_mode = (id in this.unlocked_items);
+        var modified = (id in this.modified_items);
+        var deleted = this.deleted_items_ids.includes(id);
+
+        var template = $($.parseHTML(_.template(this.html_template.html())({})));
```

Comment:
Another template that can be pre-compiled and likely benefit from placeholders.

Reply:
Fixed in ef1176accf5c411a39af7103186807ffe773720a